### PR TITLE
Deny access to disabled account but accept access to user account whe…

### PR DESCRIPTION
**Pull request by Udera**
…n domain is disabled.

Users should be able to modify their account when the domain is disabled but not when their account is disabled. The header.php automatically checks if the domain is enabled or the useraccount is still valid and generates the message. No need to pass it in line 89.

Fixes https://github.com/vexim/vexim2/issues/88

### CODE REVIEW

**Comment by rimas-kudelis** [comment]
Sorry, I guess I failed to notice: I don't think users of a disabled domain should be able to log in. Why should they?

**Comment by Udera** [comment]
I don't know the reason, but it is clearly stated here:
https://github.com/vexim/vexim2/blob/master/docs/checklist.txt#L71-L73

Perhaps the idea is to enable end users to access their mailbox even when the domain is disabled (e.g. if you move a domain to a different server, mails couldn't be delivered to the new server if the domain is still active in vexim).

File: vexim/login.php [hunk 1]
```diff
@@ -20,8 +20,8 @@
       AND username='siteadmin'
       AND users.domain_id = domains.domain_id";
   } else if ($AllowUserLogin) {
-		$query = "SELECT crypt,localpart,user_id,domain,domains.domain_id,users.admin,users.type,domains.enabled AS domainenabled FROM users,domains
-      WHERE localpart=:localpart
+		$query = "SELECT crypt,localpart,user_id,domain,domains.domain_id,users.admin,users.type,domains.enabled AS domainenabled, users.enabled AS usersenabled
```
Comment (rimas-kudelis) [inline vexim/login.php]: No need for `AS`, or at least call the field `userenabled` (without `s`).

File: vexim/login.php [hunk 2]
```diff
@@ -20,8 +20,8 @@
...
+      FROM users,domains WHERE localpart=:localpart
```
Comment (rimas-kudelis) [inline vexim/login.php]: Let's add a line-break before `WHERE`.

**Comment by rimas-kudelis** [comment]
Hm, I wonder if updating the checklist instead wouldn't make more sense. But okay, this works as well.

**Comment by Udera** [comment]
If we prefer a different behaviour, we should update the checklist. I was also wondering why you want people to access their settings when the domain is disabled. 

We need to keep this list up-to-date, it's really useful for testing.

**Comment by rimas-kudelis** [comment]
Agreed.

**Comment by Udera** [comment]
So we rather disable the login for users when the domain is disabled? For the admin as well?

What is the exact use case of domain-disabling? A customer didn't pay? Multiple abused accounts? Disable a domain, if there are temporary problems (too much spam?) For domains moving in/out?

**Comment by rimas-kudelis** [comment]
@avleen please tell us. ;)

**Comment by avleen** [comment]
I've spent a week thinking about this, and I'm not sure :-)
There is certainly value, if a domain is disabled, in allowing the users to access their mailboxes.
But, overall, I think this is more likely to present problems. For example, let's say a domain owner wants to disable access to all their mailboxes (there are many legitimate reasons for wanting to do this).. if we allow access to mailboxes in disabled domains, we just broke that expectation.

The alternative is that if we disable a domain and all mailboxes with it, people could lose access to their email. But this is a matter between the mailbox user and the domain owner.

I think the right approach, when disabling a domain, is to disable all access to accounts in that domain too.

**Comment by Udera** [comment]
Thanks for your reply. I will then create a patch that won't allow 
logins to accounts of disabled domains. If someone needs a different 
behavior, it won't be difficult for him to change it back.

As this behavior is different to previous versions, we need to put it in 
the changelogs.

**Comment by rimas-kudelis** [comment]
When fixing indentation, please do that in a separate commit.

File: vexim/login.php [hunk 3]
```diff
@@ -59,39 +60,43 @@
...
-	
+    header ('Location: site.php');
+    die();
+  }
+  if ($row['admin'] === '1') {
+    header ('Location: admin.php');
+    die();
+ }
+ if (($row['userenabled'] === '0')) {
+   header ('Location: index.php');
```
Comment (rimas-kudelis) [inline vexim/login.php]: Perhaps redirect it with a param similarly to `?domaindisabled`?

Comment (Udera) [inline vexim/login.php]: I think there is a snippet in the index.php which already adds this 
information (account disabled).

**Comment by rimas-kudelis** [comment]
eh?

**Comment by Udera** [comment]
No, it was header.php

File: vexim/login.php (hunk 3 above)
Comment (rimas-kudelis) [inline vexim/login.php]: But that is also true for disabled domains.
Should we even set the session variables in these cases?

**Comment by Udera** [comment]
I put a new pull request: https://github.com/vexim/vexim2/pull/99

**Pull request closed without merging**